strawberry/experimental/pydantic/fields.py

```diff
@@ -64,13 +60,22 @@
 }
 
 
-def get_basic_type(type_):
+def get_basic_type(type_) -> Type[Any]:
     if isinstance(type_, type):
         if issubclass(type_, pydantic.ConstrainedInt):
             return int
         if issubclass(type_, pydantic.ConstrainedStr):
             return str
 
+    if hasattr(type_, "__name__"):
+        if "ConstrainedListValue" in type_.__name__:
+            # pydantic ConstrainedListValue is created dynamically
+            # it isn't wrapped in a list, so we need to do it ourselves
+            return List[get_basic_type(type_.item_type)]  # type: ignore
+        if "ConstrainedSetValue" in type_.__name__:
+            # sets don't exist in graphql
+            return List[get_basic_type(type_.item_type)]  # type: ignore
```

Comment:
ok i'm not sure we should do this. 
whether we should help convert sets on the pydantic type to the list type.
any opinions?

Reply:
we could gate this in a configuration option :) I wouldn't do this for the time being though, it is easier to add it later than remove it if there's something wrong with it 😊

Reply:
removed the cosntrainedset support!

Reply:
yes, i have a few thoughts about how we can add configs to support conversion for pydantic... haven't gotten around to writing out the issue :D